core/dbt/contracts/graph/nodes.py

```diff
@@ -559,6 +572,16 @@ def infer_primary_key(self, data_tests: List["GenericTestNode"]) -> List[str]:
 
         return []
 
+    def format_batch_start(self, batch_start: Optional[datetime]) -> Optional[str]:
+        if batch_start is None:
+            return batch_start
+
+        return str(
+            batch_start.date()
+            if (batch_start and self.config.batch_size != BatchSize.hour)
+            else batch_start
+        )
```

Comment:
### nit
Feels somewhat weird that this is on the `ParsedNode` class instead of the [MicrobatchBuilder](https://github.com/dbt-labs/dbt-core/blob/aa23af98e57157d99b86be1571f544fe6b37c1d0/core/dbt/materializations/incremental/microbatch.py#L11) class. Just feels out of place.

Reply:
totally agree! I think at first I didn't do this because it's necessary in providers.py which didn't Added it as a static method. Good call 👍